### ISSUE TITLE
User reported issue: student who is color blind cannot differentiate atoms/molecules.

### ISSUE THREAD

**Issue by oliver-phet** [phetsims/balancing-chemical-equations#149]
> I've used your simulations for years in my classroom and I've always been so pleased with them. I recently noticed you have an accessibility tab that links to all of the simulations with accessibility features. This year I teach a student who is color blind. The Balancing Chemical Equations simulation and game has always been a helpful tool but is difficult for this particular student to use because the atoms and molecules are color coded and not labeled with any other form of identification. To this student, all of the atoms look the same and he was struggling to count how many atoms of each element he had on each side of the equation in order to balance it. PhET has always been great about updating their simulations, so I am writing to request some form of color blind accessibility for this simulation as well as other simulations that depend heavily on color coding. Thank you for your attention to this matter and for keeping chemistry inclusive for all students!

Assigning to @arouinfar for now, though this may need input by the accessibility team.

**Comment by arouinfar** [comment]
@oliver-phet it would be helpful to get more specific feedback about which atoms are hard to differentiate. I played several different levels and compared the various colorblind preview options, and I'm not immediately seeing any issues. The levels are randomly generated, however, so it's certainly possible that there are some problematic combinations.

**Comment by oliver-phet** [comment]
Reply from teacher:

> My student specifically has trouble differentiating between green and red colored atoms. He also has trouble differentiating atoms when they are only slightly different in color. For example, if a light red atom and a white atom are paired next to each other he can't see enough contrast to tell the difference between the two. The same would go for a light gray atom next to a white atom (see attached screenshot) or a light green and white atom. These were his main difficulties. If the atoms were widely different in size, he was able 
![Contrast Picture 1](https://user-images.githubusercontent.com/5090275/166004436-9da139d3-1e99-4a68-b71a-8ac64844e3a0.PNG)
![Contrast Picture 2](https://user-images.githubusercontent.com/5090275/166004439-9504d1d5-2d32-4d96-9533-c9f47c7240d7.PNG)
![Contrast Picture 3](https://user-images.githubusercontent.com/5090275/166004441-d336123a-945d-441f-b305-32ddd482ad04.PNG)
![Contrast Picture 4](https://user-images.githubusercontent.com/5090275/166004445-9723c67b-2f55-43fc-b92c-d977d35844a8.PNG)
to figure out which atoms were which most of the time. When I replayed all three rounds of the game, I noticed that chlorine (a green atom) only came up once in all three levels. If that green atom is paired with a white or red atom (as in the case of any chlorate (ClO3) molecules) he would really have a hard time. Thanks for your support!
> 
> Another example of colors he would have a hard time differentiating between would be purple and black. Not something that comes up in atomic models a lot, but something I heard him say that might be helpful!

**Comment by arouinfar** [comment]
>My student specifically has trouble differentiating between green and red colored atoms.

> I noticed that chlorine (a green atom) only came up once in all three levels. If that green atom is paired with a white or red atom (as in the case of any chlorate (ClO3) molecules) he would really have a hard time. 

I reviewed the [possible challenges](https://github.com/phetsims/balancing-chemical-equations/blob/master/js/game/model/GameFactory.js) and confirmed that chlorine (green) is never paired with oxygen (red).

> For example, if a light red atom and a white atom are paired next to each other he can't see enough contrast to tell the difference between the two. The same would go for a light gray atom next to a white atom (see attached screenshot) or a light green and white atom.

None of the colorblind previews (pasted in below) show issues with white atoms and those of other colors. Additionally, the white atoms represent Hydrogen, which is always significantly smaller than the other atoms. So even if the colors are hard to distinguish, the size difference is significant.  I'm not trying to invalidate this student's experience, but it's possible that there is some other issue. 

I'm not sure what we can do about this, short of some sort of high contrast mode like we have in Build an Atom. I'll assign to @kathy-phet to prioritize.

<img width="695" alt="image" src="https://user-images.githubusercontent.com/8419308/166063544-1632e648-9a7b-415e-b432-211bb2055157.png">
<img width="692" alt="image" src="https://user-images.githubusercontent.com/8419308/166063725-072a0408-d300-4c82-881f-5caa910df872.png">
<img width="695" alt="image" src="https://user-images.githubusercontent.com/8419308/166063757-c1881682-a418-45ba-bde5-a61ed830d7c5.png">

**Comment by pixelzoom** [comment]
Deciding on better colors would probably be easier if they could be tweaked using the Color Editor. But making the atom and molecule colors dynamic would be a big lift -- unless you were OK with the colors not updating until the Game challenge changes. It would also require redoing the memory tests in #155, since making colors dynamic has a high probability of introducing memory leaks (due to listeners added to ProfileColorProperty instances).

**Comment by pixelzoom** [comment]
This issue has been open since August 2022. It would be great to decide how to proceed.

@arouinfar's investigation is documented in https://github.com/phetsims/balancing-chemical-equations/issues/149#issuecomment-1113708902.  

The color combinations reported as problematic never occur in the sim:
> I reviewed the [possible challenges](https://github.com/phetsims/balancing-chemical-equations/blob/master/js/game/model/GameFactory.js) and confirmed that chlorine (green) is never paired with oxygen (red).

And there no problems were identified with the current colors: 
> None of the colorblind previews (pasted in below) show issues with white atoms and those of other colors. 

Since we cannot identify any colors that are problematic, I recommend that we close this issue as "won't fix".

**Comment by oliver-phet** [comment]
I'm having difficulty finding the source (what I suspect was) a phethelp email from this user. Typically, Outlook can easily find these types of emails in the phethelp inbox but I'm coming up empty.  I've also looked through our suggestion form responses, my email, and slack. 
EDIT: I found the email, but only in our old/archived phethelp gmail account. Who knows why I was unable to find in outlook. 

I've emailed the original user, to run all this by them and see if they have any comments.

**Comment by oliver-phet** [comment]
Email to user bounced (email is no longer active). Closing.

**Comment by Nancy-Salpepi** [comment]
I think I figured out what molecules the student was having difficulty with--PCl5 and PCl3 which have orange and green colors. Testing using https://webaim.org/resources/contrastchecker/ shows that these colors don't meet the 3:1 contrast ratio needed to be WCAG compliant. 

I am going to go through and check all the molecules using this tool.

**Comment by Nancy-Salpepi** [comment]
<img width="845" alt="Image" src="https://github.com/user-attachments/assets/cd01fdb2-5294-4859-9868-4a4d6cb09322" />

**Comment by pixelzoom** [comment]
To make it easier to proceed with this issue... In the above commit, I improved the nitroglycerin demo to show all molecules, screenshot below. You can customize screen background and atom stroke via the Color Editor.  

![Image](https://github.com/user-attachments/assets/9acfce35-2c03-4c58-9bae-bdf68d32dd2b)

**Comment by pixelzoom** [comment]
Problems noted with "Emulate vision deficiencies" in Chrome Dev Tools... 


**Deuteranopia (no green)**
![Image](https://github.com/user-attachments/assets/59fcbf9e-b1fd-480c-a2f9-abafa88b4a65)
![Image](https://github.com/user-attachments/assets/43f803af-f4f8-4ed1-be39-2b74b751ce5f)

**Achromatopsia (no color)***
Lots of failures where atoms of the same size can only be differentiated based on color:

![Image](https://github.com/user-attachments/assets/66c70967-a16f-45dc-9885-e23257ab49ed)

**Comment by Nancy-Salpepi** [comment]
This is a much darker green but it passes when paired with C, H and P. Thoughts?

<img width="709" alt="Image" src="https://github.com/user-attachments/assets/ef90608d-e30a-48ea-85e0-a8fbb222025d" />

**Comment by pixelzoom** [comment]
I think this may be an unsolvable problem. It will certainly be a lot of work.  I'm worried that we may start using non-standard colors for atoms, which would be even more problematic.  Color decisions we make here need to be made in nitroglycerin (common-code repo) which is used by 21 sims, so we'd have to confirm in all of those sims. And any Elements added to nitroglycerin in the future will likely make this problem come back.

So I recommend that we punt on this for BCE 2.0 release, and defer to a future release -- where we also plan to address other a11y issues like alt input and description.

@Nancy-Salpepi @kathy-phet Do you agree with deferring?

**Comment by Nancy-Salpepi** [comment]
This is a great reference for the [CPK colors](https://sciencenotes.org/molecule-atom-colors-cpk-colors/).
With today's standards it is impossible to use the exact rgb values. The goal should be to remain in that general color, eg. green, but it may be much darker or lighter than what was previously used. 

It is going to take a lot of time to figure out new values for each element that will work across all sims that use them. 
I am happy to work on this on the side, but I agree that we should move forward with BCE as is.

**Comment by pixelzoom** [comment]
BCE completed dev testing this afternoon. This is the last issue blocking creation of the 2.0 release branch.  @kathy-phet are you OK with deferring this issue?

**Comment by pixelzoom** [comment]
Deferred until a future release.

**Comment by amanda-phet** [comment]
I'd love to find a solution to this. While this is deferred right now, I wanted to put one idea down while it is fresh: 

A colorblind mode in preferences that add a non-color visual distinction between different orbs (such as a small letter, a pattern overlay, a dark stroke, etc.). While we don't want this mode to be necessary, there are good reasons to maintain a specific set of colors, and finding more than 5 colors that contrast with each other and the background that are vibrant and not "muddy" is nearly impossible and should not be attempted. I imagine we also don't want this mode for all users, because it can add a lot of clutter (so being able to turn it off is important for users who require fewer visual distractions), which is why I think a preference makes sense.

**Comment by Nancy-Salpepi** [comment]
When we revisit this sim....

The CPK coloring system is the color convention used for molecular models. (https://en.wikipedia.org/wiki/CPK_coloring). This makes the colors of atoms an essential exception. 

Many of the colors in BCE closely resemble what is listed in Jmol (software program that updated CPK colors). See https://jmol.sourceforge.net/jscolors/ for all hex/RGB values. I do see a couple of differences. 
First, the shade of yellow for Sulfur is very different. We use a dark yellow while Jmol uses a bright yellow. 

<img width="34" alt="Image" src="https://github.com/user-attachments/assets/d969a779-b928-45a5-89c0-d6c11dbad46a" /> <img width="29" alt="Image" src="https://github.com/user-attachments/assets/16806fe1-85bb-443b-9d06-993fe7faf671" />

Second, we depict Fluorine as bright yellow, but in Jmol it is a shade of green.

<img width="41" alt="Image" src="https://github.com/user-attachments/assets/6bdde0cb-e04b-452a-8e38-445ffbf6d009" /> <img width="28" alt="Image" src="https://github.com/user-attachments/assets/bb33e4d6-f025-4a01-a8be-37149a11722b" />

How close do our colors need to be to the established colors before listing them as an essential exception?
